src/query/execute_schema.c

```diff
@@ -4573,12 +4575,22 @@ do_rename_partition (MOP old_class, const char *newname)
 
   newlen = strlen (newname);
 
-  error = au_fetch_class (old_class, &smclass, AU_FETCH_READ, AU_SELECT);
+  error = au_fetch_class (old_class, &smclass, AU_FETCH_UPDATE, AU_ALTER);
   if (error != NO_ERROR)
     {
       goto end_rename;
     }
 
+  /* The pexpr format is defined in the function pt_node_to_partition_info. */
+  strncpy (expr, smclass->partition->expr, DB_MAX_PARTITION_EXPR_LENGTH);
+  expr_ptr = strstr (expr, "FROM");
+  expr_ptr = strstr (expr_ptr, "[");
```

Comment:
expr_ptr can null. it needs to check if it's null.

Reply:
Execute the do_rename_partition() function through the condition below.

``
if (is_partition == DB_PARTITIONED_CLASS)
``

pexpr of DB_PARTITION_CLASS can be NULL, but pexpr of DB_PARTITIONED_CLASS) cannot be NULL.

Reply:
in the case below
create table "from wild" ( "from river" int ):
==> SELECT [from wild] FROM [from river]

Please refer to the code below
```
// SELECT [ ~ ] FROM [ ~ ]
expr_ptr = strstr (expr, "]"); // skip column name
expr_ptr = strstr (expr_ptr + 1, "[");  // find table name position
```

Reply:
I'll add a comment.

Added above case to test case attached to jira.

Reply:
As you said, the method of finding table names in pexpr has been improved.

Reply:
at the clause expr_ptr = strstr (expr, "FROM"), expr_ptr can be null if the "FROM" is not found.

Reply:
The pexpr is created in the pt_node_to_partition_info() function.
It is not created by the user's query.

In the pt_node_to_partition_info () function, the pexpr query is fixed as "SELECT %s FROM [%s]".

The way pexpr looks up table names has changed, but pexpr can't be without "FROM".